upup/pkg/fi/cloudup/template_functions.go

```diff
@@ -77,6 +78,10 @@ func (tf *TemplateFunctions) AddTo(dest template.FuncMap) {
 
 	dest["IAMPrefix"] = tf.IAMPrefix
 	dest["IAMServiceEC2"] = tf.IAMServiceEC2
+
+	dest["AssociatePublicIP"] = func() bool {
+		return fi.BoolValue(tf.cluster.Spec.AssociatePublicIP)
```

Comment:
Oh, yes, like this :-)

So you don't need to pass in AssociatePublicIP in apply_cluster :-)